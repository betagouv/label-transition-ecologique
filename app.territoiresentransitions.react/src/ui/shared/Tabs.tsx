// recopié/adapté de react-dsfr pour gérer une prop. onChange
// https://github.com/dataesr/react-dsfr/blob/master/src/components/interface/Tabs/Tabs.js
import React, {
  cloneElement,
  useState,
  Children,
  ReactElement,
  KeyboardEvent,
} from 'react';

type TabsProps = {
  className?: string;
  children: ReactElement[];
  defaultActiveTab?: number;
  onChange: (activeTab: number) => void;
};

const Tabs = ({className, children, defaultActiveTab, onChange}: TabsProps) => {
  const [activeTab, setActiveTab] = useState(() => defaultActiveTab);

  const tabsPanel = Children.toArray(children).map((child, index) =>
    cloneElement(child as ReactElement, {
      activeTab,
      index,
    })
  );

  const handleChange = (tabIndex: number) => {
    onChange(tabIndex);
    setActiveTab(tabIndex);
  };

  const onKeyDownTab = (e: KeyboardEvent<HTMLButtonElement>, index: number) => {
    // Behavior from WAI-ARIA Authoring Practices 1.1
    // https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-19
    switch (e.key) {
      case 'ArrowRight':
        e.preventDefault();
        handleChange((index + 1) % tabsPanel.length);
        break;

      case 'ArrowLeft':
        e.preventDefault();
        handleChange(index - 1 < 0 ? tabsPanel.length - 1 : index - 1);
        break;

      case 'Home':
        e.preventDefault();
        handleChange(0);
        break;

      case 'End':
        e.preventDefault();
        handleChange(tabsPanel.length - 1);
        break;

      default:
      // do nothing => apply normal behavior
    }
  };

  const _className = `fr-tabs ${className || ''}`;
  return (
    <div className={_className}>
      <ul className="fr-tabs__list" role="tablist">
        {tabsPanel.map((element, index) => (
          <li role="presentation" key={element.props.label}>
            <button
              type="button"
              role="tab"
              id={`fr-tab-${index}`}
              aria-selected={activeTab === index ? 'true' : 'false'}
              tabIndex={activeTab === index ? 0 : -1}
              aria-controls={`fr-tabpanel-${index}`}
              className="fr-tabs__tab"
              onClick={() => handleChange(index)}
              onKeyDown={e => onKeyDownTab(e, index)}
            >
              {element.props.label}
            </button>
          </li>
        ))}
      </ul>
      {tabsPanel}
    </div>
  );
};

Tabs.defaultProps = {
  className: '',
  defaultActiveTab: 0,
};

export default Tabs;
