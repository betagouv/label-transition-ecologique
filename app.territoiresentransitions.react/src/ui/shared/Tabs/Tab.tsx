// copiÃ© depuis https://github.com/dataesr/react-dsfr/blob/master/src/components/interface/Tabs/Tab.js
import classNames from 'classnames';

type TabChildren = React.ReactNode[] | React.ReactNode | string;

interface TabProps {
  className?: string;
  children?: TabChildren;
  index?: number;
  activeTab?: number;
  label: string;
  icon?: string;
}

const Tab = ({
  className,
  index,
  activeTab,
  children,
  ...remainingProps
}: TabProps) => (
  <div
    id={`fr-tabpanel-${index}`}
    className={classNames(
      'fr-tabs__panel',
      {'fr-tabs__panel--selected': activeTab === index},
      className
    )}
    role="tabpanel"
    aria-labelledby={`fr-tab-${index}`}
    tabIndex={index}
    {...remainingProps}
  >
    {index === activeTab ? children : null}
  </div>
);

Tab.defaultProps = {
  className: '',
  icon: '',
};

export default Tab;
