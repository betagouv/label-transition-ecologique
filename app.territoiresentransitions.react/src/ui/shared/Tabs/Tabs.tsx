// copiÃ© de https://github.com/dataesr/react-dsfr/blob/master/src/components/interface/Tabs/Tabs.js
import classNames from 'classnames';
import {
  cloneElement,
  Children,
  ReactElement,
  KeyboardEvent,
  useState,
} from 'react';
//import '@gouvfr/dsfr/dist/component/tab/tab.css';

type TabsProps = {
  className?: string;
  children: (ReactElement | null)[];
  defaultActiveTab?: number;
  onChange?: (activeTab: number) => void;
};

/**
 *
 * @visibleName Tabs
 */
const Tabs = ({
  className,
  children,
  defaultActiveTab,
  onChange,
  ...remainingProps
}: TabsProps) => {
  const [activeTab, setActiveTab] = useState(() => defaultActiveTab);
  const tabsPanel = Children.toArray(children).map((child, index) =>
    cloneElement(child as ReactElement, {
      activeTab,
      index,
    })
  );

  const changeTab = (index: number) => {
    setActiveTab(index);
    if (onChange) {
      onChange(index);
    }
  };

  const onKeyDownTab = (e: KeyboardEvent<HTMLButtonElement>, index: number) => {
    // Behavior from WAI-ARIA Authoring Practices 1.1
    // https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-19
    switch (e.key) {
      case 'ArrowRight':
        e.preventDefault();
        changeTab((index + 1) % tabsPanel.length);
        break;

      case 'ArrowLeft':
        e.preventDefault();
        changeTab(index - 1 < 0 ? tabsPanel.length - 1 : index - 1);
        break;

      case 'Home':
        e.preventDefault();
        changeTab(0);
        break;

      case 'End':
        e.preventDefault();
        changeTab(tabsPanel.length - 1);
        break;

      default:
      // do nothing => apply normal behavior
    }
  };

  const _className = classNames('fr-tabs', className);
  return (
    <div className={_className} {...remainingProps}>
      <ul className="fr-tabs__list" role="tablist">
        {tabsPanel.map((element, index) => (
          <li role="presentation" key={element.props.label}>
            <button
              type="button"
              role="tab"
              id={`fr-tab-${index}`}
              tabIndex={activeTab === index ? 0 : -1}
              aria-selected={activeTab === index ? 'true' : 'false'}
              aria-controls={`fr-tabpanel-${index}`}
              className={classNames('fr-tabs__tab', element.props.icon, {
                'fr-tabs__tab--icon-left': element.props.icon,
              })}
              onClick={() => changeTab(index)}
              onKeyDown={e => onKeyDownTab(e, index)}
            >
              {element.props.label}
            </button>
          </li>
        ))}
      </ul>
      {tabsPanel}
    </div>
  );
};

Tabs.defaultProps = {
  className: '',
  defaultActiveTab: 0,
  onChange: undefined,
};

export default Tabs;
