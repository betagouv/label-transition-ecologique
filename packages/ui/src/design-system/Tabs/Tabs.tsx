import classNames from 'classnames';
import {Children, cloneElement, ReactElement, useState, useEffect} from 'react';

import {Icon} from '../Icon';

export type TabSize = 'xs' | 'sm' | 'md';

type TabsProps = {
  'data-test'?: string;
  /** Permet d'ajuster les styles du container */
  className?: string;
  /** Permet d'ajuster les styles de la liste d'onglets */
  tabsListClassName?: string;
  /** Permet d'ajuster les styles de tous les conteneurs d'onglets */
  tabPanelClassName?: string;
  /** Taille des boutons */
  size?: TabSize;
  /** Onglets */
  children: ReactElement[];
  /** Index (base 0) de l'onglet actif */
  defaultActiveTab?: number;
  /** Appelée quand l'onglet actif change */
  onChange?: (activeTab: number) => void;
};

/**
 *  Affiche un groupe d'onglets.
 */
export const Tabs = ({
  'data-test': dataTest,
  className,
  tabsListClassName,
  tabPanelClassName,
  children,
  defaultActiveTab = 0,
  size = 'md',
  onChange,
}: TabsProps) => {
  const [activeTab, setActiveTab] = useState(defaultActiveTab);

  // synchronise l'état interne
  useEffect(() => {
    setActiveTab(defaultActiveTab);
  }, [defaultActiveTab]);

  // gère le clic sur un onglet
  const handleChange = (index: number) => {
    if (index !== activeTab) {
      setActiveTab(index);
      onChange?.(index);
    }
  };

  // copie le contenu des panneaux en ajoutant les props nécessaire
  const tabsPanel = Children.toArray(children).map((child, index) =>
    cloneElement(child as ReactElement, {
      activeTab,
      index,
      className: tabPanelClassName,
    })
  );

  return (
    <div className={classNames(className)} data-test={dataTest}>
      <ul
        className={classNames(
          'inline-flex flex-wrap gap-y-6 justify-center rounded-lg bg-grey-2 p-2 gap-3 md:gap-6 w-full !list-none',
          tabsListClassName
        )}
        role="tablist"
      >
        {children.map((element, index) => {
          const isActive = activeTab === index;
          return (
            <li role="presentation" className="p-0" key={element.props.label}>
              <button
                className={classNames(
                  // styles communs
                  'px-3 py-1 font-bold w-max',
                  // variante au survol
                  'hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9',
                  {
                    // variante de taille
                    'text-md': size === 'md',
                    'text-sm': size === 'sm',
                    'text-xs': size === 'xs',
                    // variante pour l'onglet actif
                    'border border-grey-3 rounded-md shadow-button bg-white text-primary-9':
                      isActive,
                    // variante pour les onglets inactifs
                    'text-primary-10': !isActive,
                  }
                )}
                type="button"
                role="tab"
                id={`tab-${index}`}
                aria-selected={isActive ? 'true' : 'false'}
                onClick={() => handleChange(index)}
              >
                {element.props.icon && (
                  <Icon
                    icon={element.props.icon}
                    size={size}
                    className="mr-1"
                  />
                )}
                {element.props.label}
              </button>
            </li>
          );
        })}
      </ul>
      {tabsPanel}
    </div>
  );
};
