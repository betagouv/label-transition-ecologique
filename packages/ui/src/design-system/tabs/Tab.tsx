import {IconValue} from '@design-system/icons/Icon';

type TabChildren = React.ReactNode[] | React.ReactNode | string;

interface TabProps {
  /** Permet d'ajuster les styles du conteneur */
  className?: string;
  /** Contenu de l'onglet */
  children?: TabChildren;
  /** Index de l'onglet (injecté par le composant `Tabs` parent) */
  index?: number;
  /** Index de l'onglet actif (injecté par le composant `Tabs` parent) */
  activeTab?: number;
  /** Libellé de l'onglet (injecté dans le composant `Tabs`) */
  label: string;
  /** Picto optionnel affiché à gauche du libellé */
  icon?: IconValue;
}

/**
 * Affiche le contenu d'un onglet
 */
export const Tab = ({
  className,
  index,
  activeTab = 0,
  children,
  ...remainingProps
}: TabProps) =>
  index === activeTab && (
    <div
      id={`tabpanel-${index}`}
      key={`tabpanel-${index}`}
      className={className}
      role="tabpanel"
      aria-labelledby={`tab-${index}`}
      tabIndex={index}
      {...remainingProps}
    >
      {children}
    </div>
  );

Tab.defaultProps = {
  activeTab: 0,
  className: '',
};
