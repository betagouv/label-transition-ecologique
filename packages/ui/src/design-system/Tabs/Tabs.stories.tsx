import {Meta, StoryObj} from '@storybook/react';
import {action} from '@storybook/addon-actions';
import {Tabs} from './Tabs';
import {Tab} from './Tab';

const meta: Meta<typeof Tabs> = {
  component: Tabs,
  args: {
    onChange: action('onChange'),
  },
};

export default meta;

type Story = StoryObj<typeof Tabs>;

export const OngletParDefaut: Story = {
  args: {
    children: [
      <Tab label="Onglet 1">contenu onglet 1</Tab>,
      <Tab label="Onglet 2">contenu onglet 2</Tab>,
      <Tab label="Onglet 3">contenu onglet 3</Tab>,
    ],
  },
};

export const Onglet2: Story = {
  args: {
    defaultActiveTab: 1,
    children: [
      <Tab label="Onglet 1">contenu onglet 1</Tab>,
      <Tab label="Onglet 2">contenu onglet 2</Tab>,
      <Tab label="Onglet 3">contenu onglet 3</Tab>,
    ],
  },
};

export const AvecIcones: Story = {
  args: {
    defaultActiveTab: 1,
    children: [
      <Tab label="Onglet 1" icon="lock-fill">
        contenu onglet 1
      </Tab>,
      <Tab label="Onglet 2" icon="user-line">
        contenu onglet 2
      </Tab>,
      <Tab label="Onglet 3" icon="chat-1-line">
        contenu onglet 3
      </Tab>,
    ],
  },
};

export const TailleSM: Story = {
  args: {
    defaultActiveTab: 1,
    size: 'sm',
    children: AvecIcones.args.children,
  },
};

export const TailleXS: Story = {
  args: {
    defaultActiveTab: 1,
    size: 'xs',
    children: AvecIcones.args.children,
  },
};

export const AvecStyles: Story = {
  args: {
    defaultActiveTab: 2,
    tabsListClassName: 'bg-secondary-1 mb-0 rounded-none',
    tabPanelClassName: 'border-2 border-secondary-1 p-4',
    children: AvecIcones.args.children,
  },
};
