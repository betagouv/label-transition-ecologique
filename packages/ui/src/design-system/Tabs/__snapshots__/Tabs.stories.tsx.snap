// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Tabs AvecIcones smoke-test 1`] = `
<div class="overflow-hidden">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto !list-none"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="false"
      >
        <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] mr-1">
        </span>
        Onglet 1
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="true"
      >
        <span class="ri-user-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] mr-1">
        </span>
        Onglet 2
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-2"
              aria-selected="false"
      >
        <span class="ri-chat-1-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] mr-1">
        </span>
        Onglet 3
      </button>
    </li>
  </ul>
  <div id="tabpanel-1"
       role="tabpanel"
       aria-labelledby="tab-1"
       tabindex="1"
       label="Onglet 2"
       icon="user-line"
  >
    contenu onglet 2
  </div>
</div>
`;

exports[`design-system/Tabs AvecStyles smoke-test 1`] = `
<div class="overflow-hidden bg-secondary-1 mb-0 rounded-none">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto !list-none"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="false"
      >
        <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] mr-1">
        </span>
        Onglet 1
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="false"
      >
        <span class="ri-user-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] mr-1">
        </span>
        Onglet 2
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-2"
              aria-selected="true"
      >
        <span class="ri-chat-1-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] mr-1">
        </span>
        Onglet 3
      </button>
    </li>
  </ul>
  <div id="tabpanel-2"
       class="border-2 border-secondary-1 p-4"
       role="tabpanel"
       aria-labelledby="tab-2"
       tabindex="2"
       label="Onglet 3"
       icon="chat-1-line"
  >
    contenu onglet 3
  </div>
</div>
`;

exports[`design-system/Tabs Onglet2 smoke-test 1`] = `
<div class="overflow-hidden">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto !list-none"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="false"
      >
        Onglet 1
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="true"
      >
        Onglet 2
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-2"
              aria-selected="false"
      >
        Onglet 3
      </button>
    </li>
  </ul>
  <div id="tabpanel-1"
       role="tabpanel"
       aria-labelledby="tab-1"
       tabindex="1"
       label="Onglet 2"
  >
    contenu onglet 2
  </div>
</div>
`;

exports[`design-system/Tabs OngletParDefaut smoke-test 1`] = `
<div class="overflow-hidden">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto !list-none"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="true"
      >
        Onglet 1
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="false"
      >
        Onglet 2
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-2"
              aria-selected="false"
      >
        Onglet 3
      </button>
    </li>
  </ul>
  <div id="tabpanel-0"
       role="tabpanel"
       aria-labelledby="tab-0"
       tabindex="0"
       label="Onglet 1"
  >
    contenu onglet 1
  </div>
</div>
`;

exports[`design-system/Tabs TailleSM smoke-test 1`] = `
<div class="overflow-hidden">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto !list-none"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-sm text-primary-10"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="false"
      >
        <span class="ri-lock-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
        </span>
        Onglet 1
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-sm border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="true"
      >
        <span class="ri-user-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
        </span>
        Onglet 2
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-sm text-primary-10"
              type="button"
              role="tab"
              id="tab-2"
              aria-selected="false"
      >
        <span class="ri-chat-1-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
        </span>
        Onglet 3
      </button>
    </li>
  </ul>
  <div id="tabpanel-1"
       role="tabpanel"
       aria-labelledby="tab-1"
       tabindex="1"
       label="Onglet 2"
       icon="user-line"
  >
    contenu onglet 2
  </div>
</div>
`;

exports[`design-system/Tabs TailleXS smoke-test 1`] = `
<div class="overflow-hidden">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto !list-none"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-xs text-primary-10"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="false"
      >
        <span class="ri-lock-fill font-normal text-center h-3.5 w-3.5 text-[0.875rem] leading-[0.875rem] mr-1">
        </span>
        Onglet 1
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-xs border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="true"
      >
        <span class="ri-user-line font-normal text-center h-3.5 w-3.5 text-[0.875rem] leading-[0.875rem] mr-1">
        </span>
        Onglet 2
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-xs text-primary-10"
              type="button"
              role="tab"
              id="tab-2"
              aria-selected="false"
      >
        <span class="ri-chat-1-line font-normal text-center h-3.5 w-3.5 text-[0.875rem] leading-[0.875rem] mr-1">
        </span>
        Onglet 3
      </button>
    </li>
  </ul>
  <div id="tabpanel-1"
       role="tabpanel"
       aria-labelledby="tab-1"
       tabindex="1"
       label="Onglet 2"
       icon="user-line"
  >
    contenu onglet 2
  </div>
</div>
`;
